### DISCUSSION TITLE
Button calling a function from userscript

### DISCUSSION THREAD

**Discussion by ghost** [ScratchAddons/ScratchAddons#4715] [Help]
Is it possible to have a button call a function defined in the userscript? I need to do this because I need a button to call a function in the userscript which has access to vm target data.

**Comment by Norbiros** [comment]
If I understand correctly, you need to use [this](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener).
```js
document.querySelector("#obj").addEventListener("click", yourFunction);

function yourFunction() {
  console.log("click!");
}
```
*Also, you can also ask questions on our discord development server*

**Reply by ghost** [reply]
That didn't work. It stopped the errors, but it still doesn't fulfill it's purpose. I'm trying to get a sprite to go one layer up.
`function forward() {
          vm.runtime.targets[0].goForwardLayers(1);
}`
It doesn't send errors, although it doesn't work either.

**Reply by GarboMuffin** [reply]
targets[0] is the stage. Moving the stage forward a layer doesn't do anything

**Reply by ghost** [reply]
I know, but even if the index is 3 or something, it doesn't apply correctly. Do you know if Scratch Addons can even change the layer order of sprites?

**Reply by Norbiros** [reply]
Yes ScratchAddons can change layer order of elements on stage, for example look at my [rejected addon](https://github.com/ScratchAddons/ScratchAddons/pull/4399/files#diff-8ac665c04f80621f367ca43be46e01935ceb741abb8f20d3eda0d76dc78c536eR91)
This code ` vm.runtime.getSpriteTargetByName(name).goToFront();` moves sprite to top, and everything works fine.

**Reply by Norbiros** [reply]
If you want to test things with VM I recommend to use inspector (in browser) on https://turbowrap.org (it has autocomplete etc., and when you are dan just copy code to SA).

Also, your code works great for me. Can you send full addon's code (not just vm part)?

**Reply by WorldLanguages** [reply]
> Do you know if Scratch Addons can even change the layer order of sprites?

Keep in mind there's nothing Scratch Addons specific about this code you're writing - you're just running methods on the vm object.
